app/storage/src/main/java/com/fsck/k9/storage/migrations/MigrationTo84.kt

```diff
@@ -0,0 +1,54 @@
+package com.fsck.k9.storage.migrations
+
+import android.content.ContentValues
+import android.database.sqlite.SQLiteDatabase
+import com.fsck.k9.helper.map
+import com.fsck.k9.mailstore.MigrationsHelper
+
+/**
+ * Write the address fields to use ASCII 1 instead of ASCII 0 as separator.
+ * Separator was previously ASCII 0 but this caused problems with LIKE and searching.
+ */
+internal class MigrationTo84(private val db: SQLiteDatabase, private val migrationsHelper: MigrationsHelper) {
+
+    fun rewriteAddresses() {
+        val addressSets = db.rawQuery(
+            "SELECT id, to_list, cc_list, bcc_list, reply_to_list, sender_list FROM messages",
+            null
+        ).use { cursor ->
+            cursor.map {
+
+                val colVals = ArrayList<String?>(5)
+                for (i in 1..5) {
+                    colVals.add(if (it.isNull(i)) null else it.getString(i))
+                }
+                it.getLong(0) to AddressSet(colVals[0], colVals[1], colVals[2], colVals[3], colVals[4])
+            }.toMap()
+        }
+
+        var i = 0
```

Comment:
This is probably an artifact of an earlier version of the code. It's unnecessary now.